File: cucumber-json-formatter/ruby/features/step_definitions/steps.rb
Given("a passed {word}") do |word|
end

Given("a passed {word} with") do |word, arg|
end

Given("a passed {word} with text attachment {string}") do |word, attachment_text|
  embed(attachment_text, 'text/plain')
end

When("a failed {word}") do |word|
  raise "Some error"
end

When("a failed {word} with") do |word, arg|
  raise "Some error"
end

Then("a skipped {word}") do |word|
  raise "Should never be called"
end

File: cucumber-json-formatter/javascript/features/step_definitions/steps.ts
import { Given, Then, When } from 'fake-cucumber'

Given('a passed {word}', (word: string) => undefined)

Given('a passed {word} with', (word: string, arg: any) => undefined)

Given('a passed {word} with text attachment {string}', function(word: string, attachmentText: string) {
  this.attach(attachmentText, 'text/plain')
})

When('a failed {word}', () => {
  throw new Error('Some error')
})

When('a failed {word} with', () => {
  throw new Error('Some error')
})

Then('a skipped {word}', (word: string) => {
  throw new Error('Should never be called')
})
